### PR TITLE
check whether resident gauge is compatible with invert params from MILC

### LINKED ISSUE #439: MILC precision mismatch in rhmc (MILC 7.7.13 / quda 0.8.0)
QUDA internal tracking of bug report by @lcosmai https://github.com/milc-qcd/milc_qcd/issues/5

MILC 7.7.13 and probably also 7.8.0 fails in RHMC with QUDA

To reproduce build MILC for double precision and enable all QUDA acceleration for HISQ:

run
```
~/milc_qcd/ks_imp_rhmc/test$ ../su3_rhmc_hisq su3_rhmc_hisq.2.sample-in
```

error message:
```
GACTION: 5.960903e+00
ERROR: Solve precision 4 doesn't match gauge precision 8 (rank 0, host nvsocal2, /home/mathias/qudagit/lib/interface_quda.cpp:1904 in checkGauge())
       last kernel called was (name=N4quda22HeavyQuarkResidualNormI7double37double2S2_EE,volume=3x6x6x6,aux=vol=648,stride=756,precision=8)
```

### PR DESCRIPTION
**Pull request by mathiaswagner**
fixes #439

### CODE REVIEW

File: lib/interface_quda.cpp [hunk 1]
```diff
@@ -1898,6 +1898,14 @@ void MatDagMatQuda(void *h_out, void *h_in, QudaInvertParam *inv_param)
   popVerbosity();
 }
 
+namespace quda{
+bool canReuseResidentGauge(QudaInvertParam *param){
+
+  if (gaugePrecise == NULL) errorQuda("Precise gauge field doesn't exist");
```
Comment (mathiaswagner) [inline lib/interface_quda.cpp]: @maddyscientist This line is supposed to check whether a gauge field is allocated and halt if not.

**Comment by maddyscientist** [comment]
Yes, my question is whether this is the correct thing to do? If there is no gauge field allocated shouldn't we just return false?

**Comment by mathiaswagner** [comment]
Yes, that is probably ok. We will then call InvalidateGauge and this will call FreeGauge and If I looked at the right place FreeGauge will not fail if you try to fail a field that is not allocated.

**Comment by maddyscientist** [comment]
This looks good now. Merging.

**Pull request merged**